lib/matplotlib/axes/_axes.py

```diff
@@ -5093,103 +5093,140 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
-            If `norm` is an instance of `~matplotlib.colors.NoNorm`,
-            `X` must be an array of integers that index directly into
-            the lookup table of the `cmap`.
+        norm : `~matplotlib.colors.Normalize`, optional
+            If scalar data is used, the Normalize instance scales the
+            data values to the canonical colormap range [0,1] for mapping
+            to colors. By default, the data range is mapped to the
+            colorbar range using linear scaling. This parameter is ignored for
+            RGB(A) data.
 
-        vmin, vmax : scalar, optional, default: None
-            `vmin` and `vmax` are used in conjunction with norm to normalize
-            luminance data.  Note if you pass a `norm` instance, your
-            settings for `vmin` and `vmax` will be ignored.
+        vmin, vmax : scalar, optional
+            When using scalar data and no explicit *norm*, *vmin* and *vmax*
+            define the data range that the colormap covers. By default,
+            the colormap covers the complete value range of the supplied
+            data. *vmin*, *vmax* are ignored if the *norm* parameter is used.
 
-        alpha : scalar, optional, default: None
+        alpha : scalar, optional
             The alpha blending value, between 0 (transparent) and 1 (opaque).
-            The ``alpha`` argument is ignored for RGBA input data.
+            This parameter is ignored for RGBA input data.
 
-        origin : ['upper' | 'lower'], optional, default: None
+        origin : {'upper', 'lower'}, optional
             Place the [0,0] index of the array in the upper left or lower left
-            corner of the axes. If None, default to rc `image.origin`.
+            corner of the axes. The convention 'upper' is typically used for
+            matrices and images.
+            Defaults to :rc:`image.origin`.
 
-        extent : scalars (left, right, bottom, top), optional, default: None
+            Note that the vertical axes points upward for 'lower'
+            but downward for 'upper'.
+
+        extent : scalars (left, right, bottom, top), optional
             The location, in data-coordinates, of the lower-left and
-            upper-right corners. If `None`, the image is positioned such that
-            the pixel centers fall on zero-based (row, column) indices.
+            upper-right corners.
+
+            By default, pixels have a size of (1, 1). They are centered on
+            integer coordinates and their center coordinates range from 0 to
+            columns-1 horizontally and from 0 to rows-1 vertically.
+
+            Note that the relation of *top* and *bottom* to data coordinates
+            depends on *origin* because *origin* changes the direction of the
+            vertical axis.
+
+            - If origin is set to "upper", the center of the lower left
+              pixel is at (0, rows) , this corresponds to an extent
```

Comment:
I suggest something like:

 - If origin is set to "upper" then the default extents is (...)
 - If origin is set to "lower" then the default extents is (...)

If the axes auto-scales the limits than this arranges such that the value in X[0, 0] is at (0, 0) in data-coordinates in both cases.  In the case of "upper" the (0, 0) is in the upper left and the y-axis increases downward.  In the case of "lower", the (0, 0) is in the lower left and the y-axis increases upward.

Reply:
Maybe this needs a comment "having 'bottom > top' or 'left > right' may invert the drawn image"?